virttest/migration.py

```diff
@@ -0,0 +1,326 @@
+                                                   server_user, server_pwd,
+                                                   r"[\#\$]\s*$")
+            logging.info("Check VM network connectivity after migrating")
+        else:
+            logging.info("Check VM network connectivity before migration")
+            if not vm.is_alive():
+                vm.start()
+            vm.wait_for_login()
+            vm_ip[vm.name] = vm.get_address()
+            params["vm_ip_dict"] = vm_ip
+        s_ping, o_ping = utils_net.ping(vm_ip[vm.name], count=ping_count,
+                                        timeout=ping_timeout,
+                                        output_func=logging.debug,
+                                        session=server_session)
+        logging.info(o_ping)
+        if uri:
+            server_session.close()
+            vm.connect_uri = src_uri
+        if s_ping != 0:
+            if uri:
+                if "offline" in params.get("migrate_options"):
+                    logging.info("Offline Migration: %s will not responded to "
+                                 "ping as expected", vm.name)
+                    return
+            func("%s did not respond after %d sec." % (vm.name, ping_timeout))
+
+    def thread_func_migration(self, vm, desturi, options=None,
+                              ignore_status=False, virsh_opt="",
+                              extra_opts=""):
+        """
+        Thread for virsh migrate command.
+
+        :param vm: A libvirt vm instance(local or remote).
+        :param desturi: Remote host uri.
+        :param options: The options for migration command.
+        :param ignore_status: True, means no CmdError will be caught
+                              for the failure.
+                              False, means an CmdError will be caught
+                              for the failure.
+        """
+        # Migrate the domain.
+        is_error = False
+
+        try:
+            if options is None:
+                options = "--live --timeout=60"
+            stime = int(time.time())
+            self.ret = vm.migrate(desturi, option=options,
+                                  ignore_status=ignore_status,
+                                  debug=True, virsh_opt=virsh_opt,
+                                  extra=extra_opts)
+            etime = int(time.time())
+            self.mig_time[vm.name] = etime - stime
+            if self.ret.exit_status != 0:
+                logging.debug("Migration to %s returns failed exit status %d",
+                              desturi, self.ret.exit_status)
+                is_error = True
+        except process.CmdError as detail:
+            logging.error("Migration to %s failed:\n%s", desturi, detail)
+            is_error = True
+        finally:
+            if is_error is True:
+                self.RET_LOCK.acquire()
+                self.RET_MIGRATION = False
+                self.RET_LOCK.release()
```

Comment:
debug log for successful migration with migration timing (mig_time) and  other relevant details will be helpful...